### ISSUE TITLE
Provider cannot get argument values from remote state

### ISSUE THREAD

**Issue by enummela** [AviatrixSystems/terraform-provider-aviatrix#337]
The Aviatrix Terraform provider seems incapable of getting argument values from remote state (I am using S3 as my Terraform backend).

Using the example configuration below, `terraform apply` attempts to create the account without success until it eventually times out.

```
provider "aviatrix" {
  controller_ip           = data.terraform_remote_state.infrastructure.outputs.controller_public_ip
  username                = "admin"
  password                = data.aws_ssm_parameter.aviatrix_password.value
  skip_version_validation = false
}

resource "aviatrix_account" "myaccount" {
  account_name       = "myaccount"
  aws_account_number = data.aws_caller_identity.current.account_id
  aws_iam            = true
  aws_role_app       = data.terraform_remote_state.infrastructure.outputs.app_role_arn
  aws_role_ec2       = data.terraform_remote_state.infrastructure.outputs.ec2_role_arn
  cloud_type         = 1
}
```

If I create the Aviatrix account manually in the controller and run `terraform import aviatrix_account.myaccount myaccount` then an error message will complain that the `controller_ip` argument does not have a value.

The only work around I have found is to comment all of the Aviatrix provider resources, run `terraform apply` so that the remote state data source is pulled into local state, uncomment the Aviatrix provider resources, and run `terraform apply` again.

**Comment by inc1t3Ful** [comment]
Hi @enummela !

Thank you for bringing this to our attention! I will file a ticket to our team and we will definitely investigate and I will update here accordingly.

**Comment by inc1t3Ful** [comment]
Hi again, quick followup,

The **"data.terraform_remote_state.infrastructure.outputs.controller_public_ip"** seems to be referring to some controller_ip output, perhaps from the controller build module? Can you please verify if that reference has any value?

Terraform does not support referencing the module output in the provider {}

**Comment by enummela** [comment]
Hi @inc1t3Ful,

I verified that `data.terraform_remote_state.infrastructure.outputs.controller_public_ip` does indeed have a value.

It's actually not module output, but a Terraform remote state data source:
https://www.terraform.io/docs/providers/terraform/d/remote_state.html

I've used remote state with other providers, such as Kubernetes and AWS, so it seemed odd that it didn't work with the Aviatrix provider as well.

**Comment by inc1t3Ful** [comment]
We will have to investigate further. I will update again here accordingly. Thanks!

**Comment by inc1t3Ful** [comment]
Hi @enummela ,

I attempted to reproduce your issue, but I had no problems with getting argument values from remote state using the S3 as a backend.

I was able to successfully create accounts and gateways without any issues, while using S3 as the backend. Is there any more details you can offer on your set-up that could shed light onto this issue you're seeing?

**Comment by enummela** [comment]
Hi inc1t3Ful,

Thank you for investigating this. After receiving your response, I also tried replicating the issue using a pair of very small Terraform configurations and found that it worked.

Over the course of performing this test, I discovered that my provider was not locked to the correct version (2.8/2.9) based on the controller version (5.2) [[source]](https://www.terraform.io/docs/providers/aviatrix/guides/release-compatibility.html). Instead, Terraform was pulling in the latest version of the provider. I believe this may have caused the error.

Since the error was not with the provider I will close this. Cheers!
